sklearn/compose/_column_transformer.py

```diff
@@ -132,16 +133,29 @@ class ColumnTransformer(TransformerMixin, _BaseComposition):
         If True, the time elapsed while fitting each transformer will be
         printed as it is completed.
 
-    verbose_feature_names_out : bool, default=True
+    verbose_feature_names_out : bool, str or Callable[[str, str], str], default=True
```

Comment:
The rendered version of this now looks like:

![image](https://github.com/scikit-learn/scikit-learn/assets/1663864/c1fb29da-766f-42b2-ac07-1390dadafb78)

I think it makes sense to make a bullet point list for the 3 cases.

Reply:
Ok it should be good. I'm gonna try to find how to compile the doc.